enhancements/installer/credentials-management-outside-openshift-cluster.md

```diff
@@ -0,0 +1,130 @@
+
+## Proposal
+
+Currently, *mint* is a default credentials mode for OpenShift. In this mode we run OpenShift installer with an admin 
+level cloud credential. The admin credential is stored in kube-system namespace and then used by the cloud credential 
+operator to process the CredentialRequests in the cluster and create new users with fine-grained permissions.
+The customers have reported that the need to store admin credentials inside the cluster is a major disadvantage. Based 
+on the feedback, we propose to run credentials related setup outside a cluster and then start installation process in a 
+*manual* mode without providing admin credentials to the installer.
+
+### User Stories
+
+#### Story 1
+As a OpenShift cluster administrator I should be able to extract the CredentialsRequest manifests from the release image
+and create credentials in the underlying cloud provider. 
+
+#### Story 2
+As a OpenShift cluster administrator I should be able to extract the CredentialsRequest manifests from the release image 
+and create Kubernetes Secrets in to satisfy all in-cluster component's CredentialsRequests.
+
+#### Story 3
+As a OpenShift cluster administrator I should be able to install a OpenShift cluster, without providing admin credentials
+to the installer, by injecting precreated Kubernetes Secrets.
+
+### Risks and Mitigations
+
+## Design Details
+
+We intend to build an optional tool **ccoctl** which will handle credentials management of the cluster in *manual* mode. 
+The following is the set of requirements for the current prototype (for AWS). We can have something similar for other 
+platforms.
+
+* ccoctl should be able to setup Identity Provider to authenticate OpenShift components
+* ccoctl should be able to take a list of CredentialsRequests from the release image and create/update 
+  Roles/Users/ServicePrincipals/ServiceAccounts with appropriate permissions.
+* ccoctl should be able to take list of CredentialsRequests for a release image, and the Identity Provider URL to generate 
+  the objects that need to be passed to the installer for successful installation
+* ccoctl should be able to delete all the resources that it had created in the cloud
+
+We envision **ccoctl** as a recommended tool to setup credentials on new platforms but customers are free to use other 
+tools like Terraform/AWS CloudFormation to do the above mentioned tasks. Read more about AWS implementation details [here](https://github.com/openshift/cloud-credential-operator/blob/master/docs/ccoctl.md)
+
+We also plan to have a detailed documentation in place to guide new cloud providers to implement *manual* mode.
+
+### Open Questions
+
+### Test Plan
+
+We plan to have a e2e test that will externally set up a credentials management infrastructure and then kickstart 
+install in a *manual* mode.
+
+### Graduation Criteria
+
+### Upgrade / Downgrade Strategy
```

Comment:
I think this section is more about how do we roll out this new functionality. In this case it would roll out smooth as we're not changing any behavior on an existing cluster. This is more about new platforms, and future direction of CCO, should we start changing the default behavior and recommended best practice.

The notes below are relevant though and maybe could jump up to the design section.